src/main/webapp/app/exam/participate/summary/exam-participation-summary.component.ts

```diff
@@ -90,7 +136,7 @@ export class ExamParticipationSummaryComponent implements OnInit {
     }
 
     public generateLink(exercise: Exercise) {
-        if (exercise && exercise.studentParticipations && exercise.studentParticipations.length > 0) {
+        if (exercise?.studentParticipations?.[0] != undefined) {
```

Comment:
please keep the length check, otherwise it might generate an error when accessing element 0

Reply:
Same applies here, optional chaining is cleaner in my opinion. We want to access `exercise.studentParticipations[0].id` and therefore test if the element is defined. Seems to be even safer, because the if statement above only asserts if the array is non-empty, but does not check if the element is undefined (`[undefined].length → 1`). 
CC @bassner, our JS expert.

Reply:
I talked with @bassner about the optional chaining related changes and I am forwarding relevant parts of his findings here:

> Accessing an array index that is out of bounds does not throw an error in JS, but it does in Java. Your code will work, and will even be more secure as it covers the case of the array containing undefined or null at index 0 for some reason. This may be a less known fact though, so you could also simplify it to just `exercise?.studentParticipations?.length` to preserve the exact same behavior as before, and that is even shorter and similarly concise.

Reply:
Added unit tests for this simplification at commit https://github.com/ls1intum/Artemis/pull/4932/commits/9896b87f6e9769b29ac20b428dd0bfcf84dc56d2.